packages/amplify-e2e-tests/src/rds-v2-tests-common/rds-field-auth-apikey.ts

```diff
@@ -0,0 +1,619 @@
+
+      await addApiWithAllAuthModes(projRoot, { transformerVersion: 2, apiName });
+
+      await importRDSDatabase(projRoot, {
+        database,
+        engine,
+        host,
+        port,
+        username,
+        password,
+        useVpc: true,
+        apiExists: true,
+      });
+
+      const schema = /* GraphQL */ `
+        input AMPLIFY {
+          engine: String = "${engineName}"
+        }
+
+        # Api Key can't delete a record
+        # Api key can't update SSN field
+        # Api Key can't read SSN field 
+        # Api Key can create a record with all fields, but can't read SSN response
+        # Api Key subscription can't read SSN field
+        type Person1 @model @auth(rules: [{allow: public, provider: apiKey}, {allow: public, provider: iam}]) {
+          id: Int! @primaryKey
+          firstName: String
+          lastName: String
+          ssn: String @auth(rules: [{allow: public, provider: iam}, {allow: public, provider: apiKey, operations: [create]}])
+        }
+
+        # Api Key can't create a record with SSN value
+        # Api Key can create a record without a SSN value
+        # Api Key can't update a record with SSN value
+        # Api Key can't delete a record
+        type Person2 @model @auth(rules: [{allow: public, provider: apiKey}, {allow: public, provider: iam}]) {
+          id: Int! @primaryKey
+          firstName: String
+          lastName: String
+          ssn: String @auth(rules: [{allow: public, provider: iam}, {allow: public, provider: apiKey, operations: [read]}])
+        }
+
+        # Api Key can't create a record with SSN value
+        # Api Key can delete a record
+        type Person3 @model @auth(rules: [{allow: public, provider: apiKey}, {allow: public, provider: iam}]) {
+          id: Int! @primaryKey
+          firstName: String
+          lastName: String
+          ssn: String @auth(rules: [{allow: public, provider: iam}, {allow: public, provider: apiKey, operations: [delete]}])
+        }
+      `;
+      writeFileSync(rdsSchemaFilePath, schema, 'utf8');
+
+      // Enable unauthenticated access to the Cognito resource and push again
+      await enableUserPoolUnauthenticatedAccess(projRoot);
+    };
+
+    const selectionSetWithoutSSN = `
+      id
+      firstName
+      lastName
+    `;
+
+    const getQueryWithoutSSN = (name: string): string => /* GraphQL */ `
+      query Get${name}($id: Int!) {
+        get${name}(id: $id) {
+          ${selectionSetWithoutSSN}
+        }
+      }
+    `;
+
+    const listQueryWithoutSSN = (name: string): string => /* GraphQL */ `
+      query List${name}s {
+        list${name}s {
+          items {
+            ${selectionSetWithoutSSN}
+          }
+        }
+      }
+    `;
+
+    test('Model Person1 - Api Key can create a record with all fields, but SSN response is null', async () => {
+      const person = {
+        id: 1,
```

Comment:
nit: Randomizing these IDs avoids future hassle where some of these are left behind in the DB.